tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -1478,6 +1490,82 @@ public function userGetsAllApplicationsUsingTheGraphApi(string $user) {
 		$this->featureContext->setResponse($response);
 	}
 
+	/**
+	 * @When /^the administrator has given "([^"]*)" the role "([^"]*)" using the Graph API$/
+	 *
+	 * @param string $user
+	 * @param string $role
+	 *
+	 * @return void
+	 *
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public function theAdministratorHasGivenTheRoleUsingTheGraphApi(string $user, string $role): void {
+		$admin = $this->featureContext->getAdminUserName();
+		$userId = $this->featureContext->getAttributeOfCreatedUser($user, 'id');
+		$userId = $userId ?? $user;
+		$this->userGetsAllApplicationsUsingTheGraphApi($user);
+
+		$applicationEntity = ($this->featureContext->getJsonDecodedResponse($this->featureContext->getResponse()))['value'][0];
+		foreach ($applicationEntity["appRoles"] as $value) {
+			$this->appRole[$value['displayName']] = $value['id'];
+		}
+		$this->applicationId = $applicationEntity["id"];
```

Comment:
i see this `applicationId` used only to this function. is it used any where esle that you made it a class variable?

Reply:
it will be used on my follow-up PR 😸 . I forget to do clean up